lib/puppet/provider/ntp_config/cisco.rb

```diff
@@ -84,34 +111,47 @@ def validate
     return unless @resource[:source_interface]
 
     if @resource[:source_interface] =~ /\s+/
-      fail ArgumentError, "The parameter 'source_interface' must not contain" \
+      fail ArgumentError, "The parameter 'source_interface' must not contain " \
                           'any spaces'
     end
 
     return unless @resource[:source_interface] =~ /[A-Z]+/
 
-    fail ArgumentError, "The parameter 'source_interface' must not contain" \
+    fail ArgumentError, "The parameter 'source_interface' must not contain " \
                         'any uppercase characters'
   end
 
-  def munge_flush(val)
-    if val.is_a?(String) && val.eql?('unset')
-      nil
-    elsif val.is_a?(Symbol)
-      val.to_s
-    else
-      val
+  # Custom setters.
+  # The following properties are setters and cannot be handled
+  # by PuppetX::Cisco::AutoGen.mk_puppet_methods.
+  def ntp_trusted_keys
```

Comment:
Curious.  Any particular reason you decides to add the logic here at the puppet layer to remove the old and add the new instead of doing this in the node_utils setter?